app/models/auth_request.rb

```diff
@@ -4,6 +4,14 @@ class AuthRequest < ApplicationRecord
 
   validate :redirect_path_is_safe
 
+  def self.generate!(options = {})
```

Comment:
I see, so you've pushed the logic down a bit, so we do all the defaults / logic here?

Was this a general tidy up thing you noticed whilst writing these, or were the Pact tests giving you grief with the old `sign_in` method?

Reply:
Just a general tidy up.  I'd ended up with three copies of the code to create an `AuthRequest`, so I decided to put it in one place.